abipy/lumi/utils_lumi.py

```diff
@@ -0,0 +1,180 @@
+
+    n_step = len(t)
+    Lambda = abu.eV_s*0.001/(np.pi*2) * lamb  # meV to Hz
+    delta_t = t[-1] - t[0]
+
+    fourier = fft.fft(G_t * np.exp(-Lambda * np.abs(t)))
+    freq = 2 * np.pi * n_step * fft.fftfreq(G_t.size) / delta_t
+    # freq change
+    freq_2 = np.zeros(n_step)
+    fourier_2 = np.zeros(n_step, dtype=complex)
+
+    freq_2[0:n_step // 2] = freq[n_step // 2 + 1:]
+    freq_2[n_step // 2] = freq[0]
+    freq_2[n_step // 2 + 1:] = freq[1:n_step // 2 + 1]
+
+    fourier_2[0:n_step // 2] = fourier[n_step // 2 + 1:]
+    fourier_2[n_step // 2] = fourier[0]
+    fourier_2[n_step // 2 + 1:] = fourier[1:n_step // 2 + 1]
+
+    hbar_eV = abu.hbar_eVs  # in eV*s
+    En = hbar_eV * freq_2
+    E_x = En + E_zpl
+
+    sigma = w / (2.35482 * 1000)
+    gaussian = (1 / (sigma * np.sqrt(2 * np.pi))) * np.exp(-((En) ** 2 / (2 * (sigma) ** 2)))
+    A_conv = signal.fftconvolve(np.abs(fourier_2), gaussian, mode='same')
+
+    return (E_x, A_conv)
+
+
+def L_hw_help(E_x, A):
+    """     
+    Normalized Luminescence intensity (area under the curve = 1)
+    Eq. (1) of https://pubs.acs.org/doi/full/10.1021/acs.chemmater.3c00537 
+    Returns (Energy in eV, Luminescence intensity)
+
+    Args:
+        T: Temperature in K
+        lamb: Lorentzian broadening applied to the vibronic peaks, in meV
+        w: Gaussian broadening applied to the vibronic peaks, in meV
+        model: 'multi-D' for full phonon decomposition, 'one-D' for 1D-CCM PL spectrum.
+    """     
+    C = 1 / (simps(y=A * E_x ** 3, x=E_x))
+    I = C * A * E_x ** 3  # intensity prop to energy CUBE
+    return (E_x, I)
+
+
+#### Plotting functions ####
+
+
+@add_fig_kwargs
+def plot_emission_spectrum_help(x_eV,y_eV,unit,max_to_one,ax,**kwargs):
+    """     
+    Plot the Luminescence intensity, computed from the generating fct approach. 
+    
+    Args:
+        unit: 'eV', 'cm-1', or 'nm'
+        T: Temperature in K
+        lamb: Lorentzian broadening applied to the vibronic peaks, in meV
+        w: Gaussian broadening applied to the vibronic peaks, in meV
+    """     
+
+    ax, fig, plt = get_ax_fig_plt(ax=ax)
+
+#   x_eV,y_eV=self.L_hw(T=T,lamb=lamb,w=w) # in eV
+    
+    x_cm=x_eV*8065.73
+    y_cm=y_eV/8065.73
+
+    x_nm=1239.84193/x_eV
+    y_nm=y_eV*(x_eV**2/1239.84193)
+
+    if max_to_one==True:
+        y_eV = y_eV/max(y_eV)
+        y_cm = y_cm/max(y_cm)
+        y_nm = y_nm/max(y_nm)
+
+
+    if unit=='eV':
+        ax.plot(x_eV,y_eV,**kwargs)
+        ax.set_xlabel('Photon energy (eV)')
+        ax.set_ylabel(r'$L(\hbar\omega)$  (1/eV)')
+
+
+    elif unit=='cm-1':
+        ax.plot(x_cm,y_cm,**kwargs)
+        ax.set_xlabel(r'Photon energy ($cm^{-1}$)')
+        ax.set_ylabel(r'$L(\hbar\omega)$  (1/$cm^{-1}$)')
+
+    elif unit=='nm':
+        ax.plot(x_nm,y_nm,**kwargs)
+        ax.set_xlabel(r'Photon wavelength (nm))')
+        ax.set_ylabel(r'Intensity (a.u.)')
+
+    return 
```

Comment:
The decorator add_fig_kwargs expected the decorated function to return a matplotlib figure